### DISCUSSION TITLE
How to simulate protein interoperation?

### DISCUSSION THREAD

**Discussion by linonetwo** [RosettaCommons/RFdiffusion#243] [Q&A] [answered]
I tried AlphaFold3 yesterday and I find it is very easy to use, just input two protein sequence you find from uniprot.org and get the `.cif` file as result, which shows hydrogen bond between two proteins (protein interoperation).

How to achieve this with RFDifusion? I'm new to bio industry, and after several google search, I still not sure if RFDifusion can do simulation with multiple different protein, or can only output `.pdb` from a single protein's sequence.

The example in https://github.com/RosettaCommons/RFdiffusion?tab=readme-ov-file#binder-design is not very clear about this, because it only use one `.pdb` as input, instead of multiple proteins. Any comment will be helpful, thanks.

**Comment by roccomoretti** [comment, marked as answer]
RFdiffusion and AlphaFold do  different things. The goal of AlphaFold is to take a protein sequence (or multiple protein chains) and predict the 3D structure. The goal of RFdiffusion is to generate (design from scratch) protein backbones which are "native like" structures.

So RFdiffusion only takes a single pdb as a template for binder design because the typical use case there is to have just the one chain already be known and then _generate from scratch_ the other chain. If you already have a small amount of the binding partner (e.g. an important interacting peptide) RFdiffusion can also do generation with the one target chain and parts of the designed protein already specified, but when it does that it still only takes a single PDB, because you want the known part of the designed protein to already be in the desired binding configuration.

RFdiffusion isn't really set up for a use case where you have two separate PDB structures and you want to simultaneously dock them together and redesign the backbone. It's not a use case which typically comes up. If that _is_ what you're looking for, there's probably some way to mock that by adjusting the various inputs, but how to do that is likely going to depend on your exact use case, and the particular things you're hoping to achieve.

Note that if you want to do something similar to AlphaFold3, then I'd recommend [RoseTTAFold2](https://github.com/uw-ipd/RoseTTAFold2) or [RoseTTAFold All Atom](https://github.com/baker-laboratory/RoseTTAFold-All-Atom) -- RFDiffusion is based off of RoseTTAFold2, and RoseTTAFold All Atom is the AlphaFold3-like version which can handle non-protein residues. (There's also an associated [RFdiffusion All Atom](https://github.com/baker-laboratory/rf_diffusion_all_atom).

**Reply by linonetwo** [reply]
Understood, RFdiffusion's typical usecase is:

- binder design: antigen-antibody pairs, or peptide(hormone drugs)-receptor pairs
- one chain already be known: the antigen
- binding partner: hormone peptide or dominant epitope of antigen
- a single PDB which already be in the desired binding configuration: a antigen with its dominant epitope binded with some binding sites

So RFdiffusion is for people already know bindings sites, and AlphaFold3 is for people don't know where the binding site is.

I will look at projects you recommended, thanks.

**Reply by roccomoretti** [reply]
Typical use case for binder design. There's a number of other modes of RFdiffusion that aren't typically binder design problems.

I'll also note that you don't necessarily need to already know a binding site. RFdiffusion can do binder design with just the target protein present, without any specification of the binding site. It's just that in those cases you typically are doing completely free design of the binding partner (free choice of structure by RFdiffusion) rather than having a pre-specified binding partner -- though there are ways such as fold conditioning to influence the characteristics of the designed binder.
